### DISCUSSION TITLE
Code Search API for multiple-repo queries

### DISCUSSION THREAD

**Discussion by AlmostSuspense** [community/community#54835] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Is there an API for the code search that supports searching across multiple of my own repositories? I have found docs about search, but it seems to only work in a single repository and not be related to the newest version of search that's much more useful

**Comment by hrishikesh-kocharekar-dev** [comment, marked as answer]
Yes, GitHub provides a code search API that allows you to search across multiple repositories, including your own. This API is part of the GitHub REST API, which allows you to interact with GitHub programmatically.

To use the code search API, you can send a GET request to the following endpoint:
https://api.github.com/search/code

In the query string of the request, you can specify the search terms using the `q` parameter. You can also specify additional parameters to filter the search results, such as the repository or organization to search in, the file path, and the language.

Here's an example request that searches for the term "hello world" in all of your repositories:
GET https://api.github.com/search/code?q=hello+world+user:your-username&type=Code

In this example, replace "your-username" with your GitHub username. The `type` parameter is set to Code to indicate that we are searching for `code` files.

You can also authenticate your requests using a personal access token (PAT) to access private repositories or to increase your rate limit. To do this, add the following header to your request:
Authorization: Bearer <your-PAT>

Replace `<your-PAT>` with your personal access token.

**Reply by AlmostSuspense** [reply]
Thanks a lot for the in-depth explanation! 👍

**Reply by jenishdoshi** [reply]
I want to get the list of repos which has filename "Dockerfile" & "DOCKERFILE" in it.
I did the manual search which works but need only the list of repos as the no. of repos is too large. How can I get it?

**Reply by NohaZaman** [reply]
> Yes, GitHub provides a code search API that allows you to search across multiple repositories, including your own. This API is part of the GitHub REST API, which allows you to interact with GitHub programmatically.
> 
> To use the code search API, you can send a GET request to the following endpoint: https://api.github.com/search/code
> 
> In the query string of the request, you can specify the search terms using the `q` parameter. You can also specify additional parameters to filter the search results, such as the repository or organization to search in, the file path, and the language.
> 
> Here's an example request that searches for the term "hello world" in all of your repositories: GET https://api.github.com/search/code?q=hello+world+user:your-username&type=Code
> 
> In this example, replace "your-username" with your GitHub username. The `type` parameter is set to Code to indicate that we are searching for `code` files.
> 
> You can also authenticate your requests using a personal access token (PAT) to access private repositories or to increase your rate limit. To do this, add the following header to your request: Authorization: Bearer
> 
> Replace `<your-PAT>` with your personal access token.

**Reply by NohaZaman** [reply]
> Yes, GitHub provides a code search API that allows you to search across multiple repositories, including your own. This API is part of the GitHub REST API, which allows you to interact with GitHub programmatically.
> 
> To use the code search API, you can send a GET request to the following endpoint: https://api.github.com/search/code
> 
> In the query string of the request, you can specify the search terms using the `q` parameter. You can also specify additional parameters to filter the search results, such as the repository or organization to search in, the file path, and the language.
> 
> Here's an example request that searches for the term "hello world" in all of your repositories: GET https://api.github.com/search/code?q=hello+world+user:your-username&type=Code
> 
> In this example, replace "your-username" with your GitHub username. The `type` parameter is set to Code to indicate that we are searching for `code` files.
> 
> You can also authenticate your requests using a personal access token (PAT) to access private repositories or to increase your rate limit. To do this, add the following header to your request: Authorization: Bearer
> 
> Replace `<your-PAT>` with your personal access token.

**Comment by NohaZaman** [comment]
> ### Select Topic Area
> Question
> 
> ### Body
> Is there an API for the code search that supports searching across multiple of my own repositories? I have found docs about search, but it seems to only work in a single repository and not be related to the newest version of search that's much more useful
